src/main/java/com/team2502/robot2019/command/vision/GoToTargetCommand.java

```
@@ -22,33 +24,62 @@
     private double lastAbsoluteTargetAngle;
 
     private QuinticSpline lastSpline;
-    private final VisionWebsocket socket;
+    private VisionWebsocket socket;
     private boolean stop;
     private double speed;
 
-    public GoToTargetCommand() throws IOException
+    public GoToTargetCommand()
     {
-        socket = new VisionWebsocket();
         requires(Robot.DRIVE_TRAIN);
     }
 
     @Override
     protected void initialize()
     {
-        speed = Math.max(Robot.DRIVE_TRAIN.getLocEstimator().estimateAbsoluteVelocity().mag(), 1);
+        DriverStation.reportWarning("Go to target initialized", false);
+        try
+        {
+            socket = new VisionWebsocket();
+        }
+        catch(IOException e)
+        {
+            DriverStation.reportError("Failed to create socket (whoops 3 potential precursor)", e.getStackTrace());
```

I feel like we should still let the exception be thrown.... don't we not want the rest of the command to run if there is an exception or will not catching the exception completely kill the robot? I remember something along these lines with the subsystem require system I was building in `ezAuton` ... I'm surprised it would kill the whole robot though (esp since `ezAuton` runs on another thread).

---

Can we not just kill the command if the exception is thrown by setting isFinished to true?

---

well `ezAuton` uses actions and I can definitely make it just end the action without crashing the robot if I know it actually _does_ end the action, but I am waiting for confirmation if this is true.

---

I can test this Thurs

---

does anyone know if this ever occurred with ezAuton actions?